src/analysis/special_functions/gamma/beta.lean

```diff
@@ -560,3 +563,68 @@ end
 end complex
 
 end inv_gamma
+
+section doubling
+/-!
+## The doubling formula for Gamma
+
+We prove the doubling formula for arbitrary real or complex arguments, by analytic continuation from
+the positive real case. (Knowing that `Γ⁻¹` is analytic everywhere makes this much simpler, since we
+do not have to do any special-case handling for the poles of `Γ`.)
+-/
+
+namespace complex
+
+theorem Gamma_mul_Gamma_add_half (s : ℂ) :
+  Gamma s * Gamma (s + 1 / 2) = Gamma (2 * s) * 2 ^ (1 - 2 * s) * ↑(real.sqrt π) :=
+begin
+  let S := (univ : set ℂ),
```

Comment:
Instead of jumping through hoops to apply `analytic_on.eq_on_of_preconnected_of_frequently_eq`, do you think you could specialize this theorem to a global version `analytic_on.eq_of_frequently_eq`, like:
```lean
theorem eq_of_frequently_eq [connected_space 𝕜] 
  (hf : analytic_on 𝕜 f univ) (hg : analytic_on 𝕜 g univ)
  (hfg : ∃ᶠ z in 𝓝[≠] z₀, f z = g z) : f = g :=
begin
  ext1 x,
  exact eq_on_of_preconnected_of_frequently_eq hf hg is_preconnected_univ 
    (mem_univ z₀) hfg (mem_univ x),
end
```

Reply:
Makes sense, sure. I'll fork that off into a separate PR for ease of reviewing.

Reply:
In the end the changes elsewhere are very small so I've left it as part of this PR.